### PR TITLE
2021 Annual Report: SIG Architecture

### PR DESCRIPTION
**Pull request by dims**
/sig architecture

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sig-architecture/annual-report-2021.md [hunk 1]
```diff
@@ -3,80 +3,140 @@
...
-   -
+   - We care about regressions, backports, dependencies across release boundaries
+   - We care about the number of enhancements per release along with their timely graduation
    -
 
 3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
    to activities or programs that provide useful context or allow easy participation?
 
-   -
+   - TODO: Riaan
```
Comment (dims) [inline sig-architecture/annual-report-2021.md:46]: @Riaankl there's a TODO for you here.

File: sig-architecture/annual-report-2021.md [hunk 2]
```diff
@@ -87,51 +147,84 @@ In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $Y
...
+    - The team updated the repository documentation and work was started on a KEP website for easier navigation.
+    - New members joined the team: 1 new approver and 2 reviewers were added to the Enhancements Subproject. A new kep tools team with initially 4 reviewers and approvers was created and more recently had an addition of 1 new approver/reviewer. We also had 2 owners transition to become our first Emeritus approvers.
+    - We hope to keep improving the KEP process in 2022 and will be conducting a community survey to collect feedback, working on a Process KEP template and other improvements.
+- [Production Readiness](https://github.com/kubernetes/community/tree/master/sig-architecture#production-readiness-1)
+    - Production readiness review [became mandatory] for enhancements in 2021, starting with the 1.21 release cycle.
+    - We added a fourth approver, Elana Hashman, who joined the team after shadowing in 1.21.
+    - We created documentation on the [lifecycle of feature gates].
+    - We reviewed data from the 2020 Production Readiness Survey, and prepared a 2021 survey, whose [responses we analyzed]. # **TODO:** @dims to update link to point to YouTube recording
```
Comment (dims) [inline sig-architecture/annual-report-2021.md:187]: TODO for @dims

File: sig-architecture/annual-report-2021.md [hunk 3]
```diff
@@ -87,51 +147,84 @@ In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $Y
...
 
-- [ ] [README.md] reviewed for accuracy and updated if needed
-- [ ] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
+- [x] [README.md](https://github.com/kubernetes/community/blob/master/sig-architecture/README.md) reviewed for accuracy and updated if needed
+- [ ] [CONTRIBUTING.md](https://hackmd.io/GY081dMNThS16WWWB46unw) reviewed for accuracy and updated if needed
       (or created if missing and your contributor steps and experience are different or more
       in-depth than the documentation listed in the general [contributor guide] and [devel] folder.)
-- [ ] Subprojects list and linked OWNERS files in [sigs.yaml] reviewed for accuracy and updated if needed
+	  **We need to create this:**
+	- [ ] [HackMD](https://hackmd.io/GY081dMNThS16WWWB46unw) for template started - Riaan
```
Comment (sftim) [inline sig-architecture/annual-report-2021.md:215]: Is this a TODO?

Comment (Riaankl) [reply sig-architecture/annual-report-2021.md:215]: Yes, will work on it this week

File: sig-architecture/annual-report-2021.md [hunk 4]
```diff
@@ -87,51 +147,84 @@ In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $Y
...
+	    - 34 Endpoints was promoted to GA with conformance tests.
+	    - No new technical debt was incurred
+    - We moved the Conformance Office Hours Meeting from Bi-weekly to monthly in Aug 2021 to reduce the strain on attendee's calendars and increase engagement.
+    - At the start of 2021 the tracking of ineligible endpoints for conformance was moved to
+      [`ineligible_endpoints.yaml`](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/ineligible_endpoints.yaml) file.
+      This allows for more community engagement on the eligibility of endpoints for conformance before any endpoint is added to the endpoint to the 
+      [Ineligible Endpoints list](https://apisnoop.cncf.io/conformance-progress/ineligible-endpoints) through pubic pull request.
+    - Our target for 2022 is to clean up the last 51 endpoint and then move over to Conformance profiles 
+      to further increase the value added by conformance testing.
+- [Code Organization](https://github.com/kubernetes/community/tree/master/sig-architecture#code-organization-1)
```
Comment (ehashman) [inline sig-architecture/annual-report-2021.md:173]: Anything for code organization?

Comment (dims) [reply sig-architecture/annual-report-2021.md:173]: yes, this is TODO for me.

File: sig-architecture/annual-report-2021.md (hunk 3 above)
Comment (Riaankl) [reply sig-architecture/annual-report-2021.md:215]: @sftim @dims 
I update the [CONTRIBUTING.md](https://hackmd.io/GY081dMNThS16WWWB46unw) as a draft.
Please preview and propose changes as you see fit, then I will update stick it in a PR for review.

Comment (dims) [reply sig-architecture/annual-report-2021.md:215]: @Riaankl the new hackmd you pointed to, looks good.

Comment (Riaankl) [reply sig-architecture/annual-report-2021.md:215]: I will make a PR for it, then we can add a real link to the anual report.

File: sig-architecture/annual-report-2021.md [hunk 5]
```diff
@@ -87,51 +147,84 @@ In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $Y
...
+     
+- [Enhancements](https://github.com/kubernetes/community/tree/master/sig-architecture#enhancement-proposals)
```
Comment (MadhavJivrajani) [inline sig-architecture/annual-report-2021.md:175]: Would it help to mention the reading club under the enhancements sections for discoverability purposes?

Comment (dims) [reply sig-architecture/annual-report-2021.md:175]: @MadhavJivrajani yes please! propose a change and i'll gladly accept it.

File: sig-architecture/annual-report-2021.md (hunk 2 above)
Comment (johnbelamaric) [reply sig-architecture/annual-report-2021.md:187]: Survey report URL: https://datastudio.google.com/reporting/2e9c7439-202b-48a9-8c57-4459e0d69c8d/page/GC5HB

File: sig-architecture/annual-report-2021.md [hunk 6]
```diff
@@ -87,51 +147,84 @@ In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $Y
...
+    - The team had major contributions authoring and negotiating [KEP 2572 - Defining the Kubernetes release cadence](https://github.com/kubernetes/enhancements/blob/master/keps/sig-release/2572-release-cadence/).
+    - We made progress on the receipts process, but ultimately decided that it was too disruptive for the community as a new UI/UX.
+    - The team updated the repository documentation and work was started on a KEP website for easier navigation.
+    - New members joined the team: 1 new approver and 2 reviewers were added to the Enhancements Subproject. A new kep tools team with initially 4 reviewers and approvers was created and more recently had an addition of 1 new approver/reviewer. We also had 2 owners transition to become our first Emeritus approvers.
+    - We hope to keep improving the KEP process in 2022 and will be conducting a community survey to collect feedback, working on a Process KEP template and other improvements.
```
Comment (MadhavJivrajani) [inline sig-architecture/annual-report-2021.md:182]:
```suggestion
    - We hope to keep improving the KEP process in 2022 and will be conducting a community survey to collect feedback, working on a Process KEP template and other improvements.
    - In 2021, the subproject also started a [_KEP Reading Club_](https://docs.google.com/document/d/17kQZ__AfMv2D4O5PHJIKXyr9lNwYNAZjYUhvt3gRXCU/edit?usp=sharing) with the intent of creating more awareness around the features being worked on, more familiarity with the KEP process, and potentially connecting aspiring contributors with KEP authors looking for help.
      - In 2021, we've conducted [8 sessions](https://www.youtube.com/playlist?list=PL69nYSiGNLP3hHfpaik9WzlbxB43BUPJ6) of the reading club which involved discussing important KEPs such as [KEP-2221: Removing dockershim from kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2221-remove-dockershim) and [KEP-3136: Beta APIs Are Off by Default](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/3136-beta-apis-off-by-default).
      - These discussions have led to feedback that was taken by stakeholder SIGs and used to drive deeper discussions around the design of that feature, examples of such cases are [KEP-1440: kubectl events](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1440-kubectl-events) and [KEP-2258: Node service log viewer](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/2258-node-service-log-viewer/README.md).
```

File: sig-architecture/annual-report-2021.md [hunk 7]
```diff
@@ -3,80 +3,140 @@
...
+<!--
+Used this from the owners [alias file](https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES)
+Counted Unique approvers and reviewers.
+Did not count approvers again as reviewers. - Riaan
+Enhancements [owners file](https://github.com/kubernetes/enhancements/blob/master/OWNERS_ALIASES)
+- sig-architecture-approvers:
+    - dims
+    - derekwaynecarr
+    - johnbelamaric
+- api-approvers:
```
Comment (johnbelamaric) [inline sig-architecture/annual-report-2021.md:81]: Not so much annual report related, but the fact that we only have 4 API approvers is concerning.

File: sig-architecture/annual-report-2021.md (hunk 2 above)
Comment (johnbelamaric) [reply sig-architecture/annual-report-2021.md:187]: YouTube: https://youtu.be/9GM28aAyRQs

I was WAAAAAYYYYY behind on uploading....

File: sig-architecture/annual-report-2021.md (hunk 3 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md:215]: thanks @Riaankl

File: sig-architecture/annual-report-2021.md (hunk 7 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md:81]: yes definitely a bottleneck

File: sig-architecture/annual-report-2021.md (hunk 1 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md]: Fixed. added a url to https://github.com/kubernetes/community/blob/master/sig-architecture/CONTRIBUTING.md

File: sig-architecture/annual-report-2021.md (hunk 2 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md:190]: updating the report with the 2 urls

File: sig-architecture/annual-report-2021.md (hunk 3 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md:218]: linking to the merged Contributing.md

File: sig-architecture/annual-report-2021.md (hunk 4 above)
Comment (dims) [reply sig-architecture/annual-report-2021.md:173]: DONE

**Comment by dims** [comment]
@derekwaynecarr @johnbelamaric this is ready now.

thanks,
Dims

**Comment by derekwaynecarr** [comment]
@dims this looks great.

will let @johnbelamaric put a final tag.

**Comment by johnbelamaric** [comment]
Thanks everyone!

**Pull request merged**